pkg/proxy/nftables/proxier.go

```diff
@@ -221,7 +222,10 @@ func NewProxier(ipFamily v1.IPFamily,
 	} else {
 		nftablesFamily = knftables.IPv6Family
 	}
-	nft, err := knftables.New(nftablesFamily, kubeProxyTable)
+	if tableName == "" {
+		tableName = kubeProxyTable
+	}
+	nft, err := knftables.New(nftablesFamily, tableName)
```

Comment:
as you say, you probably shouldn't just use the service-proxy-name here...

We already hard-code knowledge about what characters are allowed in nftables names below (`servicePortChainNameBase`, `servicePortEndpointChainNameBase`), so I guess we could just have a function to strip/escape invalid characters from the service-proxy-name value (and prepend `"kube-proxy-"` probably?)

(in which case the argument name should be `serviceProxyName`, not `tableName`)

Reply:
No, that was a short-cut. There should be a nftables specific `tableName` parameter

Reply:
A "tableName" parameter is added in the "nftables:" section.